cmd/clusterctl/pkg/client/cluster/inventory_managementgroup.go

```diff
@@ -17,19 +17,50 @@ limitations under the License.
 package cluster
 
 import (
-	"fmt"
 	"strings"
 
 	"github.com/pkg/errors"
 	clusterctlv1 "sigs.k8s.io/cluster-api/cmd/clusterctl/api/v1alpha3"
 )
 
-func (p *inventoryClient) GetManagementGroups() ([]ManagementGroup, error) {
-	providerList, err := p.list()
-	if err != nil {
-		return nil, err
+// ManagementGroup is a group of providers composed by a CoreProvider and a set of Bootstrap/ControlPlane/Infrastructure providers
+// watching objects in the same namespace. For example, a management group can be used for upgrades, in order to ensure all the providers
+// in a management group support the same ClusterAPI version.
+type ManagementGroup struct {
```

Comment:
Adding some Utils to ManagementGroup/ManagementGroupList so It will be easier to leverage on these objects during upgrade apply